src/earn/BeforeDepositBottomSheet.tsx

```diff
@@ -0,0 +1,277 @@
+    details: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainSwapActionDescription', {
+          tokenSymbol: token.symbol,
+        })
+      : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapActionDescription', {
+          tokenSymbol: token.symbol,
+          tokenNetwork: NETWORK_NAMES[token.networkId],
+        }),
+    iconComponent: QuickActionsSwap,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_before_deposit_action_press, {
+        action: BeforeDepositActionName.CrossChainSwap,
+        ...getTokenAnalyticsProps(token),
+      })
+
+      navigate(Screens.SwapScreenWithBack, { toTokenId: token.tokenId })
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+function SwapAndDepositAction({
+  token,
+  pool,
+  forwardedRef,
+}: {
+  token: TokenBalance
+  pool: EarnPosition
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+}) {
+  const { t } = useTranslation()
+
+  const action = {
+    name: BeforeDepositActionName.SwapAndDeposit,
+    title: t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.action.swapAndDeposit'),
+    details: t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapAndDepositActionDescription', {
+      tokenSymbol: token.symbol,
+      tokenNetwork: NETWORK_NAMES[token.networkId],
+    }),
+    iconComponent: SwapAndDeposit,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_before_deposit_action_press, {
+        action: BeforeDepositActionName.SwapAndDeposit,
+        ...getTokenAnalyticsProps(token),
+      })
+
+      navigate(Screens.EarnEnterAmount, { pool, mode: 'swap-deposit' })
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+export default function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  pool,
+  hasTokensOnSameNetwork,
+  hasTokensOnOtherNetworks,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  pool: EarnPosition
+  hasTokensOnSameNetwork: boolean
+  hasTokensOnOtherNetworks: boolean
+  canAdd: boolean
+  exchanges: ExternalExchangeProvider[]
+}) {
+  const { t } = useTranslation()
+
+  const title = hasTokensOnSameNetwork
+    ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.youNeedTitle', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      })
+    : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositTitle')
+
+  return (
+    <BottomSheet
+      forwardedRef={forwardedRef}
+      title={title}
+      description={
+        !hasTokensOnSameNetwork
+          ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositDescription')
+          : undefined
+      }
+      titleStyle={styles.bottomSheetTitle}
+      testId={'Earn/BeforeDepositBottomSheet'}
+    >
+      <View style={styles.actionsContainer}>
+        {hasTokensOnSameNetwork && (
```

Comment:
Showing the swap + deposit action should also be gated by whether the pool supports swap-deposit action. We found an issue with allbridge and we won't be able to fix this before the release cut. Also, after talking to John, we decided it would be good to gate the entire swap + deposit flow in a feature gate. These changes can happen in a follow up PR though. I updated the linear issue with the updated criteria for each version of the bottom sheet.

Reply:
feature gate: https://console.statsig.com/4plizaPmWwPL21ASV4QAO0/gates/show_swap_and_deposit?newGate=true